homeassistant/components/lamarzocco/coordinator.py

```diff
@@ -0,0 +1,96 @@
+"""Coordinator for La Marzocco API."""
+from collections.abc import Callable, Coroutine
+from datetime import timedelta
+import logging
+from typing import Any
+
+from lmcloud import LMCloud as LaMarzoccoClient
+from lmcloud.exceptions import AuthFail, RequestNotSuccessful
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import CONF_MACHINE, DOMAIN
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LaMarzoccoUpdateCoordinator(DataUpdateCoordinator[None]):
+    """Class to handle fetching data from the La Marzocco API centrally."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize coordinator."""
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+        self.lm = LaMarzoccoClient(
+            callback_websocket_notify=self.async_update_listeners,
```

Comment:
Do we need to refresh at the same interval if there's a websocket that calls back when there's new data? The update coordinator supports using `async_set_updated_data` to update the data and cancel the next refresh and notify listeners.

Using that method means that as long as there's a websocket callback that is more frequent than the scan interval, there won't be any polling for new data. But if the websocket would not call back for a time longer than the scan interval, the coordinator will poll.

Reply:
we actually do, as the websocket will unfortunately only push selected data. If we abort the refresh we might see stale data for some entites.

Reply:
they are improving the websocket though it seems, once I get everything from there I'll follow up with `async_set_updated_data`

Reply:
Good!